### PR TITLE
Port Moderation functionality to TypeScript

### LINKED ISSUE #162: Port Moderation functionality to TypeScript
In an effort to refactor the frontend portion of the site, we want to move from CoffeeScript to TypeScript and in the very same step also drop stuff like jQuery to the bare minimum.

This is the sub-issue of the refactor for the moderation functionality.

**To Do:**
* [ ] Banning a user
* [ ] Removing a report
* [ ] Voting on a report
* [ ] Commenting on a report
* [ ] Setting user privileges

### PR DESCRIPTION
**Pull request by pixeldesu**
Fixes #162 

**Testing:**
* (Un)set user privileges
* (Un)ban a user
* Moderation Panel:
  * (Un)vote on reports
  * Unfold comment section
  * Comment a report
  * Character count
  * Deleting a comment
  * Deleting a report

As it was mainly used in moderation-heavy areas before, I removed `tempusdominus-bootstrap` as well, as it just broke making announcements. And subsequently I was able to get rid of the entire `legacy-moderation` script entrypoint.

It doesn't really matter (right now) to offload that script portion only to moderations as all endpoints are warden-secured in `config/routes.rb` anyway.

(Report dialog can't be ported yet as Answerbox still requires that in CoffeeScript)

### CODE REVIEW

File: app/javascript/retrospring/features/moderation/privilege.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      if data.success
+        box[0].checked = if data.checked? then data.checked else !boxChecked
+      showNotification data.message, data.success
+    error: (jqxhr, status, error) ->
+      box[0].checked = false
+      console.log jqxhr, status, error
+      showNotification translate('frontend.error.message'), false
+    complete: (jqxhr, status) ->
+      box.removeAttr "disabled"
+  */
```
Comment (raccube) [inline app/javascript/retrospring/features/moderation/privilege.ts]:
```suggestion
```

**Comment by raccube** [comment]
image
When following a user

**Review by raccube — CHANGES_REQUESTED** [review]

**Comment by pixeldesu** [comment]
> image When following a user

@raccube Following wasn't touched at all in this?

Rebuild my JS and following works fine, where did you try that, and what does the console say?

**Comment by raccube** [comment]
Ignore that - was my bad

**Review by raccube — APPROVED** [review]

**Pull request merged**